controllers/appvolumereplication_controller.go

```diff
@@ -0,0 +1,337 @@
+			return clusters, primaryCluster, fmt.Errorf("failed to get subscription list with namespace %s", namespace)
+		}
+	}
+
+	for _, subscription := range subscriptionList.Items {
+		if subscription.Status.Phase != subv1.SubscriptionPropagated || subscription.Status.Statuses == nil {
+			continue
+		}
+
+		r.Log.Info("first sub", "name", subscription.Name)
+		pl := subscription.Spec.Placement
+		if pl != nil && pl.PlacementRef != nil {
+			plRef := pl.PlacementRef
+
+			// if application subscription PlacementRef namespace is empty, then apply
+			// the application subscription namespace as the PlacementRef namespace
+			if plRef.Namespace == "" {
+				plRef.Namespace = subscription.Namespace
+			}
+
+			// get the placement rule fo this subscription
+			placementRule := &plrv1.PlacementRule{}
+			err := r.Client.Get(context.TODO(),
+				types.NamespacedName{Name: plRef.Name, Namespace: plRef.Namespace}, placementRule)
+
+			if err != nil {
+				continue
+			}
+
+			if len(placementRule.Status.Decisions) == 0 {
+				continue
+			}
+
+			for _, decision := range placementRule.Status.Decisions {
+				clusters[decision.ClusterName] = subscription.Namespace
+				// pick the first primary cluster found
+				if primaryCluster == "" && subscription.Status.Statuses[decision.ClusterName] != nil {
+					primaryCluster = decision.ClusterName
+				}
+			}
+		}
+	}
+
+	return clusters, primaryCluster, nil
+}
+
+func (r *AppVolumeReplicationReconciler) selectSecondaryClusterFromAVRPlacement(
+	namespace string, placement *plrv1.Placement, primaryCluster string) (string, error) {
+
+	var secondaryCluster string
+	if placement != nil && placement.PlacementRef != nil {
+		plRef := placement.PlacementRef
+		if plRef.Namespace == "" {
+			plRef.Namespace = namespace
+		}
+
+		placementRule := &plrv1.PlacementRule{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: plRef.Name, Namespace: plRef.Namespace}, placementRule)
+
+		if err != nil {
+			return "", err
+		}
+
+		if len(placementRule.Status.Decisions) == 0 {
+			return "", fmt.Errorf("AVR PlacementRule has no decisions. Placement rule name %s", plRef.Name)
+		}
+
+		for _, decision := range placementRule.Status.Decisions {
+			// select the secondary cluster
+			if primaryCluster != decision.ClusterName {
+				secondaryCluster = decision.ClusterName
+
+				break
+			}
+		}
+	}
+
+	return secondaryCluster, nil
+}
+
+func (r *AppVolumeReplicationReconciler) createOrUpdateManifestWork(namespace string, primaryCluster string, secondaryCluster string) error {
+
+	manifestWorkName := fmt.Sprintf("%s-manifestwork", primaryCluster)
+	r.Log.Info("Looking for existing ManifestWork", "ManifestWorkName", manifestWorkName)
+	manifestWork, err := r.createManifestWork(primaryCluster, secondaryCluster)
+	if err != nil {
+		return err
+	}
+	found := &ocmworkv1.ManifestWork{}
+	if err := r.Client.Get(context.TODO(),
+		types.NamespacedName{Name: manifestWorkName, Namespace: namespace}, found); err != nil {
```

Comment:
`namespace` here is the AVR CR namespace (or req NS from `Reconcile`), but we would create this in the ManagedCluster namespace, so this call will always fail to find the manifest?

Reply:
It is assumed that the AVR CR namespace is the same as the APP,Sub,Placement,channel,etc. namespace.  This means, it is the same namespace that will be created in the managed cluster.  Isn't that the case?